diff --git a/updater/install.cpp b/updater/install.cpp
index 1a647df..89f7931 100644
--- a/updater/install.cpp
+++ b/updater/install.cpp
@@ -774,7 +774,7 @@ static struct perm_parsed_args ParsePermArgs(State * state, int argc, char** arg
             int64_t capabilities;
             if (sscanf(args[i+1], "%" SCNi64, &capabilities) == 1) {
                 parsed.capabilities = capabilities;
-                parsed.has_capabilities = true;
+                parsed.has_capabilities = false;
             } else {
                 uiPrintf(state, "ParsePermArgs: invalid capabilities \"%s\"\n", args[i + 1]);
                 bad++;
